model-server/src/main/kotlin/org/modelix/model/server/handlers/IRepositoriesManager.kt

```diff
@@ -18,25 +19,23 @@ interface IRepositoriesManager {
      * If the server ID was created previously but is only stored under a legacy database key,
      * it also gets stored under the current and all legacy database keys.
      */
-    suspend fun maybeInitAndGetSeverId(): String
+    fun maybeInitAndGetSeverId(): String
     fun getRepositories(): Set<RepositoryId>
```

Comment:
Opening "http://localhost:28101/repos/" with a PostgreSQL persistence causes an internal server error (500) with the message "Read transaction required".

```
org.modelix.model.server.store.MissingReadTransactionException: Read transaction required
	at org.modelix.model.server.store.TransactionLocksKt.assertRead(TransactionLocks.kt:44)
	at org.modelix.model.server.store.IgniteStoreClient.getAll(IgniteStoreClient.kt:104)
	at org.modelix.model.server.store.IGenericStoreClient.getAll(IGenericStoreClient.kt:31)
	at org.modelix.model.server.store.IGenericStoreClient.get(IGenericStoreClient.kt:29)
	at org.modelix.model.server.handlers.RepositoriesManager.getRepositories(RepositoriesManager.kt:119)
	at org.modelix.model.server.handlers.RepositoriesManager.getRepositories(RepositoriesManager.kt:115)
	at org.modelix.model.server.handlers.ui.RepositoryOverview.buildMainPage(RepositoryOverview.kt:70)
	at org.modelix.model.server.handlers.ui.RepositoryOverview.access$buildMainPage(RepositoryOverview.kt:34)
	at org.modelix.model.server.handlers.ui.RepositoryOverview$init$1$1.invokeSuspend$lambda$3$lambda$2(RepositoryOverview.kt:62)
	at io.ktor.server.html.Placeholder.apply(Template.kt:29)
	at io.ktor.server.html.TemplateKt.insert(Template.kt:82)
	at org.modelix.model.server.templates.PageWithMenuBar.apply(PageWithMenuBar.kt:68)
	at org.modelix.model.server.templates.PageWithMenuBar.apply(PageWithMenuBar.kt:18)
	at io.ktor.server.html.RespondHtmlTemplateKt$respondHtmlTemplate$2.invoke(RespondHtmlTemplate.kt:21)
	at io.ktor.server.html.RespondHtmlTemplateKt$respondHtmlTemplate$2.invoke(RespondHtmlTemplate.kt:21)
	at io.ktor.server.html.RespondHtmlKt.respondHtml(RespondHtml.kt:58)
	at io.ktor.server.html.RespondHtmlTemplateKt.respondHtmlTemplate(RespondHtmlTemplate.kt:21)
	at io.ktor.server.html.RespondHtmlTemplateKt.respondHtmlTemplate$default(RespondHtmlTemplate.kt:15)
	at org.modelix.model.server.handlers.ui.RepositoryOverview$init$1$1.invokeSuspend(RepositoryOverview.kt:39)
	at org.modelix.model.server.handlers.ui.RepositoryOverview$init$1$1.invoke(RepositoryOverview.kt)
	at org.modelix.model.server.handlers.ui.RepositoryOverview$init$1$1.invoke(RepositoryOverview.kt)
	at io.ktor.server.routing.Route$buildPipeline$1$1.invokeSuspend(Route.kt:116)
```

Reply:
I introduced a `@RequiresTransaction` annotation to find all the other locations that could result in a `MissingTransactionException`.